src/openswathalgo/include/OpenMS/OPENSWATHALGO/ALGO/Scoring.h

```diff
@@ -54,6 +54,18 @@ namespace OpenSwath
 
 
     //@{
+    typedef std::pair<unsigned int, unsigned int> pos2D;
+    /// Simple hash function for Scoring::pos2D
+    struct pair_hash 
+    {
+      template <class T1, class T2>
+      std::size_t operator () (const std::pair<T1,T2> &p) const {
+        auto h1 = std::hash<T1>{}(p.first);
+        auto h2 = std::hash<T2>{}(p.second);
+        return h1 ^ h2;  
```

Comment:
Would it make sense to take one of the “established” hash combine functions? E.g. does this collide more often

Reply:
@cbielow  We could then add this to our concept or math classes? 
```
inline void hash_combine(std::size_t& seed) { }

template <typename T, typename... Rest>
inline void hash_combine(std::size_t& seed, const T& v, Rest... rest) {
    std::hash<T> hasher;
    seed ^= hasher(v) + 0x9e3779b9 + (seed<<6) + (seed>>2);
    hash_combine(seed, rest...);
}
```

Reply:
yep sure.
I'll open a ticket so we do not forget.
Student projects are coming up here, so I can put that on the agenda.
I'd suggest to keep this PR as is and deal with hashing in one go in a new PR.